homeassistant/components/sunsynkweb/sensor.py

```diff
@@ -0,0 +1,174 @@
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_grid_import",
+        translation_key="acc_grid_import",
+        suggested_display_precision=2,
+        agg_func=max,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy exported to the grid."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_grid_export",
+        translation_key="acc_grid_export",
+        suggested_display_precision=2,
+        agg_func=max,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy injected into the batteries."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_battery_charge",
+        translation_key="acc_battery_charge",
+        suggested_display_precision=2,
+    ),
+    SunsynkEntitySensorDescription(  # ""Total energy provided by the batteries."""
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        key="acc_battery_discharge",
+        translation_key="acc_battery_discharge",
+        suggested_display_precision=2,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SunsynkConfigEntry,
+    async_add_entities: AddEntitiesCallback,  # noqa: F821
+) -> None:
+    """Set up sensor devices."""
+    coordinator: SunsynkUpdateCoordinator = entry.runtime_data
+
+    async_add_entities(
+        SunSynkApiSensor(coordinator, description)
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class SunSynkApiSensor(CoordinatorEntity[SunsynkUpdateCoordinator], SensorEntity):
+    """Parent class for sunsynk api exposing sensors.
+
+    The sensors expose the sum of the power across all inverters
+    for plants that have more than one inverter.
+    State of charge is normally shared between inverters, so this will take
+    the maximum of the state of charge.
+    """
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self, coordinator, description: SunsynkEntitySensorDescription
+    ) -> None:
+        """Initialise the common elements for sunsynk web api sensors."""
+        CoordinatorEntity.__init__(self, coordinator, context=coordinator)
+        self.coordinator = coordinator
+        self._attr_unique_id = (
+            f"{description.key}_{sum(p.id for p in coordinator.cache.plants)}"
+        )
+        self.entity_id = f"sensor.sunsynk_{description.key}"
+
+        self.entity_description: SunsynkEntitySensorDescription = description
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                (DOMAIN, str(sum(plant.id for plant in self.coordinator.cache.plants))),
```

Comment:
I discussed this with one of the maintainers, and he also agreed that we should not aggregate over all plants by default. Integrations should display the "raw" data and it's up to the user to aggregrate.

Reply:
Thanks. I've update the MR to do this, but now I am struggling with the prettier pre-commit hook and the dev container; I unfortunately can't afford to spend much more time on this, so will bow out of contributing this change.
thanks a lot for all the useful feedback and for the time you spent helping me.